File: ros1bookcode-master-rw/chapt5/chapt5_ws/src/demo_cpp_tf/src/static_tf_broadcaster.cpp
#include <ros/ros.h>
#include <geometry_msgs/TransformStamped.h>
#include <tf2/LinearMath/Quaternion.h>
#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
#include <tf2_ros/static_transform_broadcaster.h>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "static_tf_broadcaster");
  static tf2_ros::StaticTransformBroadcaster broadcaster;
  geometry_msgs::TransformStamped transform;
  transform.header.stamp = ros::Time::now();
  transform.header.frame_id = "base_link";
  transform.child_frame_id = "target_point";
  transform.transform.translation.x = 1.0;
  transform.transform.translation.y = 2.0;
  transform.transform.translation.z = 0.0;
  tf2::Quaternion quat; quat.setRPY(0, 0, 0);
  transform.transform.rotation = tf2::toMsg(quat);
  broadcaster.sendTransform(transform);
  ros::spin();
  return 0;
}

File: ros1bookcode-master-rw/chapt5/chapt5_ws/src/demo_python_tf/demo_python_tf/static_tf_broadcaster.py
#!/usr/bin/env python3
import rospy
import tf2_ros
from geometry_msgs.msg import TransformStamped
from tf.transformations import quaternion_from_euler

if __name__ == '__main__':
    rospy.init_node('static_tf_broadcaster')
    br = tf2_ros.StaticTransformBroadcaster()
    transform = TransformStamped()
    transform.header.stamp = rospy.Time.now()
    transform.header.frame_id = 'base_link'
    transform.child_frame_id = 'bottle_link'
    transform.transform.translation.x = 0.3
    transform.transform.translation.y = 0.0
    transform.transform.translation.z = 0.5
    q = quaternion_from_euler(0, 0, 0)
    transform.transform.rotation.x = q[0]
    transform.transform.rotation.y = q[1]
    transform.transform.rotation.z = q[2]
    transform.transform.rotation.w = q[3]
    br.sendTransform(transform)
    rospy.spin()
